### DISCUSSION TITLE
No events received since 4 days ago.

### DISCUSSION THREAD

**Discussion by LarsLautrup** [fwestenberg/reolink_dev#469] [General]
HA hasn't received any events since 4 days ago. It could be at the same time as HA was upgraded to 2022.3.3, but I'm not sure.

I have a couple of automations, triggered by the "Person detected"-event and they have stopped working. I've changed it to the general "Motion detection"-event, but that is not working either and it looks like no events is fired at all.
I have tried reinstalling the integration and the camera, but with no luck.

Is there anybody here, that are experiencing the same issues?

EDIT: I found this in the log. All cameras have problems.

**This error originated from a custom integration.

Logger: custom_components.reolink_dev.base
Source: custom_components/reolink_dev/base.py:395
Integration: Reolink IP camera (documentation, issues)
First occurred: 05:18:17 (15 occurrences)
Last logged: 05:24:17

Host 192.168.0.107 error renewing the Reolink subscription
Host 192.168.0.105 error renewing the Reolink subscription
Host 192.168.0.108 error renewing the Reolink subscription**

I haven't changed anything and I'm not using https (as far as I know ...)

/Lars.

**Comment by xannor** [comment]
The integration relies on ONVIF notify subscriptions for push notifications (its the only "automatic" method that reolink provides for 3rd party, but their implementation has issues.) This however is dependent on the camera being able to connect back to the home assistant instance. Recent changes in HA have brought up issues if you use a mixed https/non-https external/internal url setup on your HA instance. There was a recent commit to address this, but I believe it would generate a log message stating so.

First you may want to verify you are only using one method to connect to the server (i.e. either https or plain http) or just dont have internal/external url set in the general settings. Also if you had issues with authentication, the cameras can get overwhelmed and it is best to tel lthem to reboot to flush their security and also restart HA to keep things in sync.

-- correction --: I found the link about https/http (#462) and it looks like the Reolink cameras have https issues (I never use it on them and dont use https on my HA install)

**Reply by LarsLautrup** [reply]
I'm not using https at all, but I'm not sure if something turned it on somewhere for some reason 4-5 days ago...
I found this in the log and all cameras has the same issue:

This error originated from a custom integration.

Logger: custom_components.reolink_dev.base
Source: custom_components/reolink_dev/base.py:395
Integration: Reolink IP camera (documentation, issues)
First occurred: 05:18:17 (15 occurrences)
Last logged: 05:24:17

Host 192.168.0.107 error renewing the Reolink subscription
Host 192.168.0.105 error renewing the Reolink subscription
Host 192.168.0.108 error renewing the Reolink subscription

I have no idea, what the cause of the errors could be ...

/Lars.

**Reply by xannor** [reply]
That error means that it cannot renew the ONVIF subscription. Try logging into the camera web interface, and telling it to reboot, then restart your HA instance. The ONVIF implementation in these cameras is flakey at best, and they get easily confused.

**Reply by LarsLautrup** [reply]
> That error means that it cannot renew the ONVIF subscription. Try logging into the camera web interface, and telling it to reboot, then restart your HA instance. The ONVIF implementation in these cameras is flakey at best, and they get easily confused.

Tried that,  but it still doesnt work.

I was getting desperate, so I removed all cameras, removed the integration from HACS and rebooted everything.
When adding the first camera, I discovered, that HTTPS seems to be required!? Whenever I uncheck HTTPS and changes the portnumber to 80, the camera was not found. It's only found when there's HTTPS is checked and the port number is 443 - and then the same error occurs in the log. :-(   Getting a stream from the camera works though, but that has never been a problem.

![image](https://user-images.githubusercontent.com/17698826/158172664-81cb3af9-c7b9-4812-8696-bb353652328f.png)

The cameras are not using https.

/Lars.

**Reply by xannor** [reply]
Not being able to turn off https seems like a bug to me, though it should not be impacting the cameras on if as that is push not pull. At this point I think you need to convert this to an issue, also enable debugging, but before posting anything scrub the logs to remove auth info because it will get captured.

**Reply by LarsLautrup** [reply]
Will do. Thanks for all your help! :-)

/Lars.

**Comment by LarsLautrup** [comment]
Hi! 
I just updated HA to 2022.3.5 and everything started working again. 
I have absolutely no idea why and what the problem was. The release notes for 2022.3.5 dont have any fixes, that has anything to do with this issue, as I see it ...

Now, I just hope, that it stays that way :-)

/Lars.

**Comment by cpainchaud** [comment]
May be my fix in version 0.51? :)

**Reply by LarsLautrup** [reply]
Of course! That could be it! I updated to 0.51 just before updating HA to 2022.3.5 
Thanks!!! 👍 

/Lars.
